gems/quilt_rails/README.md

```diff
@@ -300,6 +300,67 @@ function App() {
 export default App;
 ```
 
+##### Example: sending headers from Rails controller
+
+```ruby
+class ReactController < ApplicationController
+  include Quilt::ReactRenderable
+
+  def index
+    render_react(headers: { 'x-custom-header': 'header-value-a' })
+  end
+end
+```
+
+You will need to serialize the result of the useRequestHeader hook in order for it to persist to the client
+
+```tsx
+// app/ui/foundation/CustomUniversalProvider.tsx
+import {createContext} from 'react';
+import {createUniversalProvider} from '@shopify/react-universal-provider';
+
+export const CustomContext = createContext<string | null>(null);
```

Comment:
Assuming we don't already have a broken version of this as per my above question, I wonder if it's worth providing a `UniversalHeaderProvider` as part of `@shopify/react-network` that handles serializing headers that are grabbed by `useRequestHeader` during SSR.

Reply:
(out of scope for this PR, but could be a follow up)

Reply:
@TheMallen I think a UniversalHeaderProvider would be a good idea, I implemented a basic version for the warehouse project but it probably makes sense to add to react-network if `useRequestHeader` will need to be serialized.